File: ai/src/main/java/org/conductoross/conductor/ai/models/AudioGenRequest.java
package org.conductoross.conductor.ai.models;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@EqualsAndHashCode(callSuper = true)
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class AudioGenRequest extends LLMWorkerInput {
    private String text;
    private String voice;
    @Builder.Default private double speed = 1.0;
    @Builder.Default private String responseFormat = "mp3";
    @Builder.Default private int n = 1;
}

File: ai/ai/src/models/AudioGenRequest.ts
import { LLMWorkerInput } from './LLMWorkerInput.js';

export class AudioGenRequest extends LLMWorkerInput {
  text?: string;
  voice?: string;
  speed: number = 1.0;
  responseFormat: string = 'mp3';
  n: number = 1;
}
